controllers/cf/cf_pod_spec.go

```diff
@@ -0,0 +1,318 @@
+		}
+		baseContainer.Args = currentContainer.Args
+		// containers[*].command
+		if len(baseContainer.Command) > 0 {
+			return nil, NewConfigError("containers[0].command")
+		}
+		baseContainer.Command = currentContainer.Command
+		// containers[*].env
+		if len(baseContainer.Env) > 0 {
+			return nil, NewConfigError("containers[0].env")
+		}
+		baseContainer.Env = currentContainer.Env
+		// containers[*].image
+		if baseContainer.Image != "" {
+			return nil, NewConfigError("containers[0].image")
+		}
+		baseContainer.Image = currentContainer.Image
+		// containers[*].imagePullPolicy
+		if baseContainer.ImagePullPolicy != "" {
+			return nil, NewConfigError("containers[0].imagePullPolicy")
+		}
+		baseContainer.ImagePullPolicy = currentContainer.ImagePullPolicy
+		// containers[*].livenessProbe
+		if baseContainer.LivenessProbe == nil {
+			baseContainer.LivenessProbe = factoryContainer.LivenessProbe
+		}
+		// containers[*].name
+		if baseContainer.Name != "" {
+			return nil, NewConfigError("containers[0].name")
+		}
+		baseContainer.Name = factoryContainer.Name
+		// containers[*].ports
+		if len(baseContainer.Ports) > 0 {
+			return nil, NewConfigError("containers[0].ports")
+		}
+		baseContainer.Ports = currentContainer.Ports
+		// containers[*].readinessProbe
+		if baseContainer.ReadinessProbe == nil {
+			baseContainer.ReadinessProbe = factoryContainer.ReadinessProbe
+		}
+		// containers[*].resources
+		if len(baseContainer.Resources.Limits) == 0 &&
+			len(baseContainer.Resources.Requests) == 0 {
+			baseContainer.Resources = factoryContainer.Resources
+		}
+		// containers[*].workingDir
+		if baseContainer.WorkingDir != "" {
+			return nil, NewConfigError("containers[0].workingDir")
+		}
+		baseContainer.WorkingDir = currentContainer.WorkingDir
+	} else {
+		base.Containers = make([]ar.ApicurioRegistrySpecDeploymentPodSpecPreviewContainer, len(factory.Containers))
+		for i, c := range factory.Containers {
+			cc, err := ConvertFromContainer(&c)
+			if err != nil {
+				return nil, err
+			}
+			base.Containers[i] = *cc
+		}
+	}
+	// ephemeralContainers
+	if len(base.EphemeralContainers) > 0 {
+		return nil, NewConfigError("ephemeralContainers")
+	}
+	base.EphemeralContainers = current.EphemeralContainers
+	// imagePullSecrets
+	if len(base.ImagePullSecrets) > 0 {
+		return nil, NewConfigError("imagePullSecrets")
+	}
+	base.ImagePullSecrets = current.ImagePullSecrets
+	// initContainers
+	if len(base.InitContainers) > 0 {
+		return nil, NewConfigError("initContainers")
+	}
+	base.InitContainers = current.InitContainers
+	// terminationGracePeriodSeconds
+	if base.TerminationGracePeriodSeconds == nil {
+		base.TerminationGracePeriodSeconds = factory.TerminationGracePeriodSeconds
+	}
+	// tolerations
+	if len(base.Tolerations) > 0 {
+		return nil, NewConfigError("tolerations")
+	}
+	base.Tolerations = current.Tolerations
+	// volumes
+	for _, v := range factory.Volumes {
+		common.SetVolume(log, &base.Volumes, &v)
```

Comment:
We should merge all the volumes coming from `podTemplate` and from the desired Deployment.